pdfrw/objects/pdfstring.py

```diff
@@ -1,74 +1,507 @@
+between this module, the tokenizer, and the pdfwriter.
+
+tokenizer string handling
+--------------------------
+
+As far as the tokenizer and its clients such as the pdfreader are concerned,
+the PdfString class must simply be something that it can instantiate by
+passing a string, that doesn't compare equal (or throw an exception when
+compared) to other possible token strings.  The tokenizer must understand
+enough about the syntax of the string to successfully find its beginning
+and end in a stream of tokens, but doesn't otherwise know or care about
+the data represented by the string.
+
+pdfwriter string handling
+--------------------------
+
+The pdfwriter knows and cares about two attributes of PdfString instances:
+
+  - First, PdfString objects have an 'indirect' attribute, which pdfwriter
+    uses as an indication that the object knows how to represent itself
+    correctly when output to a new PDF.  (In the case of a PdfString object,
+    no work is really required, because it is already a string.)
+  - Second, the PdfString.encode() method is used as a convenience to
+    automatically convert any user-supplied strings (that didn't come
+    from PDFs) when a PDF is written out to a file.
+
+pdfstring handling
+-------------------
+
+The code in this module is designed to support those uses by the
+tokenizer and the pdfwriter, and to additionally support encoding
+and decoding of PdfString objects as a convenience for the user.
+
+Most users of the pdfrw library never encode or decode a PdfString,
+so it is imperative that (a) merely importing this module does not
+take a significant amount of CPU time; and (b) it is cheap for the
+tokenizer to produce a PdfString, and cheap for the pdfwriter to
+consume a PdfString -- if the tokenizer finds a string that conforms
+to the PDF specification, it will be wrapped in a PdfString object,
+and if the pdfwriter finds an object with an indirect attribute, it
+simply calls str() to ask it to format itself.
+
+Encoding and decoding are not actually performed very often at all,
+compared to how often tokenization and then subsequent concatenation
+by the pdfwriter are performed.  In fact, versions of pdfrw prior to
+0.4 did not even support Unicode for this function.  Encoding and
+decoding can also easily be performed by the user, outside of the
+library, and this might still be recommended, at least for encoding,
+if the visual appeal of encodings generated by this module is found
+lacking.
+
+
+PdfString.decode()
+~~~~~~~~~~~~~~~~~~~
+
+Decoding strings can be tricky, but is a bounded process.  Each
+properly-encoded encoded string represents exactly one output string,
+with the caveat that is up to the caller of the function to know whether
+he expects a Unicode string, or just bytes.
+
+The decode function defaults to producing a Unicode string, but will
+produce bytes if the to_bytestring parameter is set True.  Unicode
```

Comment:
Probably because it's automatic and useful, without having to remember the semantics of the optional parameter and the default...  Anyway, yeah, I left `decode` in, doing what it always did, and enhanced `encode`.